wdl-grammar/src/v1/lint/document_preamble.rs

```diff
@@ -0,0 +1,272 @@
+//! Incorrect document preamble
+
+use std::collections::VecDeque;
+
+use nonempty::NonEmpty;
+use pest::iterators::Pair;
+use wdl_core::concern::code;
+use wdl_core::concern::lint;
+use wdl_core::concern::lint::Rule;
+use wdl_core::concern::lint::Tag;
+use wdl_core::concern::lint::TagSet;
+use wdl_core::concern::Code;
+use wdl_core::file::Location;
+use wdl_core::Version;
+
+use crate::v1;
+
+/// Detects improper document preamble.
+/// Checks for whitespace, comments, and version declaration placement.
+#[derive(Debug)]
+pub struct DocumentPreamble;
+
+impl<'a> DocumentPreamble {
+    /// Generates a lint warning for an improperly placed version
+    /// declaration.
+    fn misplaced_version(&self, location: Location) -> lint::Warning
+    where
+        Self: Rule<&'a Pair<'a, v1::Rule>>,
+    {
+        // SAFETY: this error is written so that it will always unwrap.
+        lint::warning::Builder::default()
+            .code(self.code())
+            .level(lint::Level::Low)
+            .tags(self.tags())
+            .subject("Improperly placed version declaration")
+            .body(
+                "The version declaration must be the first line of a WDL document or one blank \
+                 line after header comments.",
```

Comment:
This message needs to be word-smithed. The phrasing confuses me.

Reply:
This is what I went with. Let me know if you think this is better.

```
The version declaration must be the first line of a WDL document or immediately
following exactly one blank line and any header comments.
```

Reply:
Good except for the use of `header comments`. It should say `preamble comments` so that we remain consistent with our terms.

Reply:
Good catch!